subprojects/maven/src/main/java/org/gradle/api/publication/maven/internal/deployer/BaseMavenDeployer.java

```diff
@@ -51,6 +51,7 @@ public BaseMavenDeployer(PomFilterContainer pomFilterContainer, ArtifactPomConta
     protected MavenPublishAction createPublishAction(File pomFile, File metadataFile, LocalMavenRepositoryLocator mavenRepositoryLocator) {
         MavenWagonDeployAction deployAction = new MavenWagonDeployAction(pomFile, metadataFile, getJars());
         deployAction.setLocalMavenRepositoryLocation(mavenRepositoryLocator.getLocalMavenRepository());
+        deployAction.produceLegacyMavenMetadata();
```

Comment:
Why should they be moved here?

Reply:
I don't understand your question.

Reply:
I mean, looks like you were moving `session.getConfigProperties().put("maven.metadata.legacy", "true");` from `AbstractMavenPublishAction` constructor to here, why is it?

Reply:
Because `AbstractMavenPublishAction` is used for all Maven publishing, via the `Upload` task _and_ via the `maven-publish` plugin. We don't want to change the default for `Upload`, but we want to disable legacy publishing via the `@Incubating` `maven-publish` plugin. So I moved this down to a common type used by `Upload`, but not `maven-publish`.